### PR TITLE
[improvement] Add TracedChannel

### PR DESCRIPTION
**Pull request by markelliot**
## Before this PR
Dialogue does not trace any requests.

## After this PR
Users of Dialogue may use a TracedChannel in the channel hierarchy in order to trace specific channel executions with a named operation.

### CODE REVIEW

File: dialogue-core/src/main/java/com/palantir/dialogue/core/TracedChannel.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import com.palantir.tracing.Tracers;
+
+public final class TracedChannel implements Channel {
+    private final Channel delegate;
+    private final String operationName;
+
+    TracedChannel(Channel delegate, String operationName) {
```
Comment (carterkozak) [inline dialogue-core/src/main/java/com/palantir/dialogue/core/TracedChannel.java:30]: I think this either needs a public factory, or the class should be package private

Comment (markelliot) [reply dialogue-core/src/main/java/com/palantir/dialogue/core/TracedChannel.java:30]: made package private to match other *Channel classes.

**Comment by carterkozak** [comment]
lgtm 👍

**Comment by markelliot** [comment]
unsure of the procedure for cutting a release here

**Comment by markelliot** [comment]
also looks like your thumbs up wasn't enough here, @carterkozak

**Comment by ferozco** [comment]
👍

**Comment by carterkozak** [comment]
I don't appear to have perms, sorry

**Comment by ferozco** [comment]
Fixed it so that @carterkozak should be able to approve moving forward

**Pull request merged**